File: HackerRank/simpleArraySum.js
/* https://www.hackerrank.com/challenges/simple-array-sum/problem */

/* https://codeburst.io/javascript-arrays-finding-the-minimum-maximum-sum-average-values-f02f1b0ce332 */

function simpleArraySum(ar) {
    return ar.reduce((accumulator, value) => accumulator + value, 0)
}

console.log(simpleArraySum([1,2,3,4,5,6]));

File: Solutions/HackerRank/simpleArraySum.c
/* https://www.hackerrank.com/challenges/simple-array-sum/problem */

#include<stdio.h>
#include<stdlib.h>

int simpleArraySum(int ar_count, int* ar);

int main(){
	
	int array[] = {1,2,3,4,5,6};
	int lenght = sizeof(array)/sizeof(array[0]);
	
	printf("%d\n", simpleArraySum(lenght, array));
	
	return 0;	
}

int simpleArraySum(int ar_count, int* ar) {
    
    int i, sum=0;
    
    for(i=0; i<ar_count; i++){
    	
    	sum += ar[i];
	}
	
	return sum;
}
